src/chainparamsbase.cpp

```diff
@@ -63,41 +60,31 @@ class CBaseRegTestParams : public CBaseChainParams
         strDataDir = "regtest";
     }
 };
-static CBaseRegTestParams regTestParams;
-
-/*
- * Unit test
- */
-class CBaseUnitTestParams : public CBaseMainParams
-{
-public:
-    CBaseUnitTestParams()
-    {
-        strDataDir = "unittest";
-    }
-};
-static CBaseUnitTestParams unitTestParams;
 
-static CBaseChainParams* pCurrentBaseParams = 0;
+static Container<CBaseChainParams> cGlobalChainBaseParams;
 
 const CBaseChainParams& BaseParams()
 {
-    assert(pCurrentBaseParams);
-    return *pCurrentBaseParams;
+    return cGlobalChainBaseParams.Get();
 }
 
-void SelectBaseParams(const std::string& chain)
+CBaseChainParams* CBaseChainParams::Factory(const std::string& chain)
```

Comment:
What is the reason for using `std::string` instead of a enum for chain identification? We still could have something like `static std::string ChainParams::ToString(enum chain)`